pymc3/tests/test_sampling.py

```diff
@@ -121,6 +121,17 @@ def test_iter_sample(self):
             for i, trace in enumerate(samps):
                 assert i == len(trace) - 1, "Trace does not have correct length."
 
+    def test_sample_does_not_modify_start(self):
+        start_dict = {"X0_mu": 25}
+        with self.model:
+            X0_mu = pm.Lognormal("X0_mu", mu=np.log(0.25), sd=0.10)
+            trace = pm.sample(
+                tune=50,
+                draws=100,
+                start=start_dict,
+            )
+        assert len(start_dict) == 1
```

Comment:
perhaps just `assert start_dict == {"X0_mu": 25}`?

also, is it possible to make a test case which hits the other branch (i.e. where there is a list of dicts)?

Reply:
I think, `assert len(start_dict) == 1` would also work as `sample` method just add the `transformed_RV` to the dictionary. Anyway I will replace that.
 Even ,I was just thinking of adding the test for list of dict too.

Reply:
A list of `start` dicts can be passed to initialize each chain on a different parameterset.
You can switch the step method to `pm.Metropolis`, because it is much faster to initialize. Also reduce to `tune=5, draws=10, chains=3` or so to speed up the test a bit.
The parameters of the distribution are not actually important for this test case.

Reply:
Small suggestion: you can add a comment at the beginning of the test function referring to the original issue for some context